### PR TITLE
Include all boundary cells in GlobalNy

### PR DESCRIPTION
**Pull request by johnomotani**
The `GlobalNx`, `GlobalNy` variables include boundary cells, but previously `GlobalNy` only included boundary cells from one divertor. If a second divertor boundary is present, the guard cells from there should also be included. This change makes `GlobalNy` consistent with `getGlobalYIndex` in the sense that on the last processor `getGlobalYIndex(LocalNy) == GlobalNy` always.

The lines setting the global grid sizes are moved below those getting the separatrix/X-point locations just so that setting `GlobalNy` can use `jyseps2_1` and `jyseps1_2` - the other lines are not changed.

### CODE REVIEW

**Comment by johnomotani** [comment]
The Travis tests are not completing because they time out when `test-initial` is trying to run on 3 processors (the 1 and 2 processor cases run and pass). The test only takes a few seconds on my laptop which only has 2 physical cores, and looks like it used to take ~30s on Travis (looking at the last commit on `master`). I also don't think this PR should particularly affect the test. Has Travis changed something??

**Comment by ZedThree** [comment]
Not to my knowledge. The same tests work fine on other builds. I'll have a look on my machine now

**Comment by ZedThree** [comment]
Works on my machine too, so must be something about travis. Will investigate further this afternoon

File: src/mesh/impls/bout/boutmesh.cxx [hunk 1]
```diff
@@ -170,6 +162,17 @@ int BoutMesh::load() {
   Mesh::get(jyseps2_2, "jyseps2_2", ny - 1);
   Mesh::get(ny_inner, "ny_inner", jyseps2_1);
 
+  // Set global grid sizes
+  GlobalNx = nx;
```
Comment (ZedThree) [inline src/mesh/impls/bout/boutmesh.cxx]: This needs to be set before the calls to `Mesh::get(ixseps...)` on L157,158

File: src/mesh/impls/bout/boutmesh.cxx [hunk 2]
```diff
@@ -170,6 +162,17 @@ int BoutMesh::load() {
...
+  GlobalNy = ny + 2 * MYG;
+  if (jyseps1_2 != jyseps2_1) {
+    GlobalNy += 2*MYG;
+  }
+  GlobalNz = nz;
+
+  if (2 * MXG >= nx)
```
Comment (ZedThree) [inline src/mesh/impls/bout/boutmesh.cxx]: I know this was already there, but it's identical to the check on L152, so let's delete it

File: src/field/globalfield.cxx [hunk 3]
```diff
@@ -239,8 +241,10 @@ int GlobalField2D::msg_len(int proc) const {
 
 ///////////////////////////////////////////////////////////////////////////////////////////
 
-GlobalField3D::GlobalField3D(Mesh *m, int proc) : GlobalField(m, proc, m->GlobalNx, m->GlobalNy-2*m->ystart, m->LocalNz), 
-  data_valid(false) {
+GlobalField3D::GlobalField3D(Mesh *m, int proc)
+  : GlobalField(m, proc, m->GlobalNx,
+                m->GlobalNy-mesh->numberOfYBoundaries()*2*m->ystart, m->LocalNz),
```
Comment (ZedThree) [inline src/field/globalfield.cxx]: I got a warning about `mesh` being uninitialised here -- I guess `m` is correct.

**Comment by ZedThree** [comment]
Please can you add a section at the top of [`CHANGELOG.md`](https://github.com/boutproject/BOUT-dev/tree/next/CHANGELOG.md) like so:

```markdown
# Change Log

## [v5.0.0-alpha](https://github.com/boutproject/BOUT-dev/tree/next) 
[Full Changelog](https://github.com/boutproject/BOUT-dev/compare/v4.3.0...next)

### Breaking changes

- `BoutMesh::GlobalNy` now counts points from all
  y-boundaries. Previously it only counted points from one boundary
  [\#1829](https://github.com/boutproject/BOUT-dev/pull/1829)
```

**Pull request merged**